builtin/providers/aws/resource_aws_security_group_rule.go

```diff
@@ -595,3 +602,15 @@ func setFromIPPerm(d *schema.ResourceData, sg *ec2.SecurityGroup, rule *ec2.IpPe
 
 	return nil
 }
+
+// Validates that either 'cidr_blocks', 'self', or 'source_security_group_id' is set
+func validateAwsSecurityGroupRule(d *schema.ResourceData) error {
+	_, blocksOk := d.GetOk("cidr_blocks")
+	_, sourceOk := d.GetOk("source_security_group_id")
+	_, selfOk := d.GetOk("self")
+	if !blocksOk && !sourceOk && !selfOk {
+		return fmt.Errorf(
+			"One of ['cidr_blocks', 'self', 'source_security_group_id'] must be set to create an AWS Security Group Rule")
```

Comment:
I think `prefix_list_ids` need to be added to this special club, see `TestAccAWSSecurityGroupRule_PrefixListEgress`